### ISSUE TITLE
VM Settings validation blocks custom key value pairs

### ISSUE THREAD

**Issue by rajujith** [apache/cloudstack#7700]
##### ISSUE TYPE

  * Improvement Request
 
##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.17,4.18 mostly related to the new UI. 
~~~

##### SUMMARY

Currently, the VM settings in the UI validate (not in API) the parameters. It allows only a known/predefined list of parameters in the drop-down available. However, this is restrictive for advanced users and more use cases. Custom parameter/key-value pairs could be allowed if required to display a message that it is not in the predefined list and the user is expected to make sure it is supported etc. 

##### STEPS TO REPRODUCE

~~~
1. Stop a VM
2. Try adding a new parameter via the VM settings tab.
~~~

##### EXPECTED RESULTS

~~~
VM settings UI could allow custom key-value pairs. 
~~~

##### ACTUAL RESULTS

~~~
VM settings UI doesn't allow custom key-value pairs. 

~~~

![Screenshot 2023-06-28 at 11 38 11 AM](https://github.com/apache/cloudstack/assets/5229147/ebda81dc-8238-4faa-82cd-e8bd8c931574)

**Comment by weizhouapache** [comment]
fixed by #7793

**Comment by weizhouapache** [comment]
thanks @DaanHoogland for fixing it
